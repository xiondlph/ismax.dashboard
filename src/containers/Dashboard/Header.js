import React, { Component } from 'react'
import classNames from 'classnames'

export default class Dashboard extends Component {
    toggleSidebarState() {
        const { sidebar } = this.props

        if (sidebar === 'full') this.props.setSidebarState('mini')
        if (sidebar === 'mini') this.props.setSidebarState('full')
    }

    render() {
        const { sidebar } = this.props

        return <div className='dashboard-header'>
            <div className={classNames('dashboard-header-logo', `dashboard-header-logo_${sidebar}`)}>
                <span className='dashboard-header-logo-icon'></span>
                ISMAX icsystem
            </div>
            <div className='dashboard-header-toolbar'>
                <a onClick={::this.toggleSidebarState}><span className='fa fa-navicon'></span></a>
            </div>
        </div> 
    }
}