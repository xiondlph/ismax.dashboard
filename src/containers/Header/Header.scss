.dashboard-header {
  @include box-shadow(0, 1px, 2px, 0, rgba(0, 0, 0, 0.2));
  width: 100%;
  height: $header-height;
  line-height: $header-height;
  display: flex;
  flex-direction: row;
  z-index: 2;

  &-logo {
    background-color: #e05d6f;
    transition: width 0.5s;
    white-space: nowrap;
    overflow: hidden;
    font-size: 16px;
    height: 100%;
    width: 250px;
    color: #fff;

    &_mini {
      width: 64px;
    }

    &-icon {
      font-family: icsystem;
      display: inline-block;
      text-align: center;
      font-size: 25px;
      width: 64px;
      float: left;

      &:before {
        content: "\e60c";
      }
    }
  }

  &-toolbar {
    padding: 20px;
    flex: 1;

    &-btn {
      color: $dashboard-text-color-gray;
      text-align: center;
      line-height: 24px;
      cursor: pointer;
      display: block;
      height: 24px;
      width: 24px;

      &-toggle {
        float: left;
      }

      &-logout {
        float: right;
      }
    }

    &-label {
      color: $dashboard-text-color-gray;
      display: inline-block;
      margin-right: 10px;
      line-height: 24px;
      float: right;
      height: 24px;
    }

    &-photo {
      display: inline-block;
      margin-right: 10px;
      line-height: 24px;
      color: cadetblue;
      font-size: 24px;
      height: 24px;
      float: right;
    }
  }
}
