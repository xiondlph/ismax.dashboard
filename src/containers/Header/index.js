import React, { Component } from 'react'
import classNames from 'classnames'

import './Header.scss'

export default class Header extends Component {
    toggleSidebarState() {
        const { sidebar } = this.props

        if (sidebar === 'full') this.props.setSidebarState('mini')
        if (sidebar === 'mini') this.props.setSidebarState('full')
    }

    render() {
        const { sidebar } = this.props

        return (
            <div className='dashboard-header'>
                <div className={classNames('dashboard-header-logo', `dashboard-header-logo_${sidebar}`)}>
                    <span className='dashboard-header-logo-icon'></span>
                    ISMAX icsystem
                </div>
                <div className='dashboard-header-toolbar'>
                    <a className='dashboard-header-toolbar-btn dashboard-header-toolbar-btn-toggle' onClick={::this.toggleSidebarState}>
                        <span className='dashboard-header-toolbar-btn-toggle-icon fa fa-navicon'></span>
                    </a>

                    <a className='dashboard-header-toolbar-btn dashboard-header-toolbar-btn-logout'>
                        <span className='dashboard-header-toolbar-btn-toggle-icon fa fa-sign-out'></span>
                    </a>

                    <a className='dashboard-header-toolbar-label'>
                        admin@ismax.ru
                    </a>

                    <a className='dashboard-header-toolbar-photo'>
                        <span className='dashboard-header-toolbar-photo-icon fa fa-user-circle-o'></span>
                    </a>
                </div>
            </div>
        )
    }
}