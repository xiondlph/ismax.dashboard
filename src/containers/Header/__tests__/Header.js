import React from 'react'
import renderer from 'react-test-renderer'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Header from '../Header'

configure({ adapter: new Adapter() });

test('Проверка Header снапшота', () => {
    const component = renderer.create(
        <Header />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('Проверка смены состояния Sidebar', () => {
    const
        sidebar = 'mini',
        setSidebarState = () => {
            header.setProps({ sidebar: 'full' })
        },
        header = shallow(
            <Header sidebar={sidebar} setSidebarState={setSidebarState}/>
        );

    expect(header.find('.dashboard-header-logo').exists()).toEqual(true)
    expect(header.find('.dashboard-header-logo').hasClass('dashboard-header-logo_mini')).toEqual(true)

    header.find('.dashboard-header-toolbar a').simulate('click');
    expect(header.find('.dashboard-header-logo').hasClass('dashboard-header-logo_full')).toEqual(true)
})
