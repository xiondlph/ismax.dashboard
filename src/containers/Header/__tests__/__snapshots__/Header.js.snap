// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка Header снапшота 1`] = `
<div
  className="dashboard-header"
>
  <div
    className="dashboard-header-logo dashboard-header-logo_undefined"
  >
    <span
      className="dashboard-header-logo-icon"
    />
    ISMAX icsystem
  </div>
  <div
    className="dashboard-header-toolbar"
  >
    <a
      onClick={[Function]}
    >
      <span
        className="fa fa-navicon"
      />
    </a>
  </div>
</div>
`;
